### DISCUSSION TITLE
Preset not creating pasckage.json?

### DISCUSSION THREAD

**Discussion by ari-hellomolly** [barrel/shopify-vite#150] [Q&A]
Hi,

I work at an ecommerce company and will be rebuilding one of our sites in shopify from scratch. I plan on using Volt + Vite on top of the Dawn theme as a starting point. I've been testing the installation guides in the documentation and something I've noticed is that if I use the preset approach, while it perfectly scaffolds a starting point for the volt plugin, it doesn't include a package.json with dependencies or scripts. Is this the expected behavior or is the expectation to create a package.json with npm init (or in my case, pnpm init) before running the preset cli command? The documentation specifically mentions "Once you have installed the preset, you do not need to follow the rest of the installation guide." however it only talks about creating a package.json in the ensuing installation guide.

What's the expected workflow for someone using the preset, once they've initialised a fresh shopify theme and run the preset cli commands? There are "next steps" that get echoed in the terminal once the preset command has been run, however it references running scripts with npm which can't be done until there is a package.json with scripts & dependencies included:

![Screenshot 2024-08-19 at 11 43 06 AM](https://github.com/user-attachments/assets/700a1594-50ca-4b4e-99e8-50026f50679d)

**Comment by montalvomiguelo** [comment]
Hey @ari-hellomolly, The preset should install all the necessary files to start using Vite and Shopify Vite Plugin in a fresh Shopify theme, including the package.json file.


https://github.com/user-attachments/assets/8fdd9f7b-d6ee-4a74-b89d-d8b3b6dc5cda

**Reply by joshuafredrickson** [reply]
Performing the exact same commands as above gets me the following. No `package.json`.

```bash
$ git status

On branch main
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .gitignore
	modified:   layout/password.liquid
	modified:   layout/theme.liquid

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.shopifyignore
	frontend/
	jsconfig.json
	postcss.config.cjs
	tailwind.config.cjs
	vite.config.js

no changes added to commit (use "git add" and/or "git commit -a")
```
